src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -0,0 +1,345 @@
+}
+
+@observer
+export default class SurvivalChart extends React.Component<ISurvivalChartProps, {}> {
+
+    @observable tooltipModel: any;
+    private isTooltipHovered: boolean = false;
+    private tooltipCounter: number = 0;
+    private alteredLegendText = 'Cases with Alteration(s) in Query Gene(s)';
+    private unalteredLegendText = 'Cases without Alteration(s) in Query Gene(s)';
+    private svgContainer:any;
+    private svgsaver = new SvgSaver();
+
+    constructor(props: ISurvivalChartProps) {
+        super(props);
+        this.tooltipMouseEnter = this.tooltipMouseEnter.bind(this);
+        this.tooltipMouseLeave = this.tooltipMouseLeave.bind(this);
+        this.downloadSvg = this.downloadSvg.bind(this);
+        this.downloadPng = this.downloadPng.bind(this);
+    }
+
+    @computed get sortedAlteredPatientSurvivals(): PatientSurvival[] {
+        return this.props.alteredPatientSurvivals.sort((a, b) => a.months - b.months);
+    }
+
+    @computed get sortedUnalteredPatientSurvivals(): PatientSurvival[] {
+        return this.props.unalteredPatientSurvivals.sort((a, b) => a.months - b.months);
+    }
+
+    @computed get alteredEstimates(): number[] {
+        return getEstimates(this.sortedAlteredPatientSurvivals);
+    }
+
+    @computed get unalteredEstimates(): number[] {
+        return getEstimates(this.sortedUnalteredPatientSurvivals);
+    }
+
+    @computed get logRank(): number {
+        return calculateLogRank(this.sortedAlteredPatientSurvivals, this.sortedUnalteredPatientSurvivals);
+    }
+
+    private tooltipMouseEnter(): void {
+        this.isTooltipHovered = true;
+    }
+
+    private tooltipMouseLeave(): void {
+        this.isTooltipHovered = false;
+        this.tooltipModel = null;
+    }
+
+    private downloadSvg() {
+        this.svgsaver.asSvg(this.svgContainer.firstChild, this.props.fileName + '.svg');
+    }
+
+    private downloadPng() {
+        this.svgsaver.asPng(this.svgContainer.firstChild, this.props.fileName + '.png');
+    }
+
+    public render() {
+
+        const events = [{
+            target: "data",
+            eventHandlers: {
+              onMouseOver: () => {
+                return [
+                  {
+                    target: "data",
+                    mutation: () => ({ active: true })
+                  },
+                  {
+                    target: "labels",
+                    mutation: (props:any) => { 
+                        this.tooltipModel = props;
+                        this.tooltipCounter++;
+                    }
+                  }
+                ];
+              },
+              onMouseOut: () => {
+                return [
+                  {
+                    target: "data",
+                    mutation: () => ({ active: false })
+                  },
+                  {
+                    target: "labels",
+                    mutation: async () => {
+                        await sleep(100);
```

Comment:
is this like a wrapper for setTimeout?  I just want to make sure it's not some kind of looping that is actually sychronous

Reply:
@alisman it's a function I created in src/shared/lib/TimeUtils.ts, a new file present in this PR.

Reply:
even though it makes the function execution "synchronous", await is asynchronous https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

Reply:
i know that await is async.  i think i got hung up on "sleep" because in my experience sleep means the processor locks.   i think maybe "wait" might be better.  but it's not a big deal.  i just wanted to make sure it wasn't some thing like for (i < 1trillion) because people have written such things!